ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -1237,6 +1243,39 @@ public void getIndexSegmentFilesForLogSegmentTest() {
     }
   }
 
+  /**
+   * Tests {@link PersistentIndex#close()} can correctly cancel the scheduled persistor task and makes sure no persistor
+   * is running background after index closed.
+   * @throws StoreException
+   * @throws InterruptedException
+   */
+  @Test
+  public void closeIndexToCancelPersistorTest() throws StoreException, InterruptedException {
+    state.index.close();
+    // re-initialize index by using mock scheduler (the intention is to speed up testing by using shorter period)
+    ScheduledThreadPoolExecutor scheduler = (ScheduledThreadPoolExecutor) Utils.newScheduler(1, false);
+    ScheduledThreadPoolExecutor mockScheduler = Mockito.spy(scheduler);
+    AtomicReference<ScheduledFuture> persistorTask = new AtomicReference<>();
+    MockIndexPersistor mockPersistor = new MockIndexPersistor();
+    mockPersistor.invokeCountDown = new CountDownLatch(1);
+    doAnswer(invocation -> {
+      Object[] args = invocation.getArguments();
+      persistorTask.set(mockScheduler.scheduleAtFixedRate(mockPersistor, 0, 50, TimeUnit.MILLISECONDS));
+      return persistorTask.get();
+    }).when(mockScheduler)
+        .scheduleAtFixedRate(any(Runnable.class), anyLong(), anyLong(), any(TimeUnit.SECONDS.getClass()));
```

Comment:
Agree, let me try if we can use better identifiers. (My initial thought is `IndexPersistor.class` might need to be exposed, because currently it is `private`)